packages/tokens/src/rwa/identity_registry_storage/storage.rs

```diff
@@ -0,0 +1,600 @@
+/// }
+/// ```
+use soroban_sdk::{
+    contracttype, panic_with_error, vec, Address, Env, Map, String, Symbol, TryFromVal, Val, Vec,
+};
+
+use crate::rwa::identity_registry_storage::{
+    emit_country_data_event, emit_identity_modified, emit_identity_stored, emit_identity_unstored,
+    CountryDataEvent, IRSError, IDENTITY_EXTEND_AMOUNT, IDENTITY_TTL_THRESHOLD,
+    MAX_COUNTRY_ENTRIES,
+};
+
+/// ISO 3166-1 numeric country code
+pub type CountryCode = u32;
+
+/// Represents the type of identity holder
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum IdentityType {
+    Individual,
+    Organization,
+}
+
+/// Represents different types of country relationships for individuals
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum IndividualCountryRelation {
+    /// Country of residence
+    Residence(CountryCode),
+    /// Country of citizenship
+    Citizenship(CountryCode),
+    /// Country where funds originate
+    SourceOfFunds(CountryCode),
+    /// Tax residency (can differ from residence)
+    TaxResidency(CountryCode),
+    /// Custom country type for future extensions
+    Custom(Symbol, CountryCode),
+}
+
+/// Represents different types of country relationships for organizations
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum OrganizationCountryRelation {
+    /// Country of incorporation/registration
+    Incorporation(CountryCode),
+    /// Countries where organization operates
+    OperatingJurisdiction(CountryCode),
+    /// Tax jurisdiction
+    TaxJurisdiction(CountryCode),
+    /// Country where funds originate
+    SourceOfFunds(CountryCode),
+    /// Custom country type for future extensions
+    Custom(Symbol, CountryCode),
+}
+
+/// Unified country relationship that can be either individual or organizational
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum CountryRelation {
+    Individual(IndividualCountryRelation),
+    Organization(OrganizationCountryRelation),
+}
+
+/// A country data containing the country relationship and optional metadata
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CountryData {
+    /// Type of country relationship
+    pub country: CountryRelation,
+    /// Optional metadata (e.g., visa type, validity period)
+    pub metadata: Option<Map<Symbol, String>>,
+}
+
+/// Complete identity profile containing identity type and country data
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct IdentityProfile {
+    pub identity_type: IdentityType,
+    pub countries: Vec<CountryData>,
+}
+
+/// Storage keys for the data associated with Identity Storage Registry.
+#[contracttype]
+pub enum IRSStorageKey {
+    /// Maps account address to identity address
+    Identity(Address),
+    /// Maps an account to its complete identity profile
+    IdentityProfile(Address),
+}
+
+// ################## QUERY STATE ##################
+
+/// Retrieves the stored identity for a given account.
```

Comment:
I'm confused on the difference between `Account Address` and `Identity Address`. 

If I missed the relative inline comments somewhere else, it's ok. 

If not, can we explain it here also?